src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1353,18 +1364,29 @@ public void DeleteRevision(DocumentsOperationContext context, Slice key, string
                 // but we don't want to mess up the order of events so the delete revision etag we use is negative
                 revisionEtag = _documentsStorage.GenerateNextEtagForReplicatedTombstoneMissingDocument(context);
             }
-            CreateTombstone(context, key, revisionEtag, collectionName, changeVector, lastModifiedTicks);
+
+            CreateTombstone(context, key, revisionEtag, collectionName, changeVector, lastModifiedTicks, fromReplication);
         }
 
         private unsafe void CreateTombstone(DocumentsOperationContext context, Slice keySlice, long revisionEtag,
-            CollectionName collectionName, string changeVector, long lastModifiedTicks)
+            CollectionName collectionName, string changeVector, long lastModifiedTicks, bool fromReplication = false)
         {
             var newEtag = _documentsStorage.GenerateNextEtag();
 
             var table = context.Transaction.InnerTransaction.OpenTable(TombstonesSchema, RevisionsTombstonesSlice);
             if (table.VerifyKeyExists(keySlice))
                 return; // revisions (and revisions tombstones) are immutable, we can safely ignore this
 
+            if (DatabaseRecord.GetSupportedFeaturesFor(_database.SupportedFeatures).ThrowRevisionKeyTooBigFix &&
```

Comment:
would like to have something like `_database.SupportedFeatures.ThrowRevisionKeyTooBigFix`